sklearn/inspection/tests/test_partial_dependence.py

```diff
@@ -644,12 +648,18 @@ def test_partial_dependence_dataframe(estimator, preprocessor, features):
 
 @pytest.mark.parametrize(
     "features, expected_pd_shape",
-    [(0, (3, 10)),
```

Comment:
Even though the API spec of `partial_dependence()` only allows for `array-like of {int, str}` for `feature` parameter (https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/inspection/_partial_dependence.py#L248), we seem to be allowing `int`, `str`, `list of str`, `list of int`, and `boolean mask`. 
This unnecessarily complicated initialization of `is_categorical` parameter when it's `None` at https://github.com/scikit-learn/scikit-learn/pull/18298/files#diff-0c8623ee957e0256d29ac53830094281R488-R489.
Do we need to keep supporting this undocumented behaviour?

Reply:
The use case was supported before but the documentation is misleading. Basically we should support a sequence of {int, str} or tuple of {int, str}:

https://github.com/scikit-learn/scikit-learn/pull/12599/files#diff-f6ab89354a7e7254ec59bb2e192b0ec2R211-R218

So the test here create a scalar or a tuple (actually this is a list) which will be added in another list within the test.

Maybe changing the documentation with the following would be less ambiguous: `features : array-like of {int, str} or tuple of 2 {int, str}`